From 6a1f3cfadbc5b9a0469804ee06839ef048c959a7 Mon Sep 17 00:00:00 2001
From: Juan Quintela <quintela@redhat.com>
Date: Thu, 4 Mar 2010 23:11:26 -0300
Subject: [PATCH 04/42] savevm.c: fix warning with _FORTIFY_SOURCE

RH-Author: Juan Quintela <quintela@redhat.com>
Message-id: <b3d8c62bbdeae7019a8caf3ad4f3d73d5ad18972.1267743950.git.quintela@redhat.com>
Patchwork-id: 7531
O-Subject: [PATCH 04/32] savevm.c: fix warning with _FORTIFY_SOURCE
Bugzilla: 567099
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Amit Shah <amit.shah@redhat.com>
RH-Acked-by: Marcelo Tosatti <mtosatti@redhat.com>

From: Kirill A. Shutemov <kirill@shutemov.name>

CC    savevm.o
cc1: warnings being treated as errors
savevm.c: In function 'file_put_buffer':
savevm.c:342: error: ignoring return value of 'fwrite', declared with attribute warn_unused_result
make: *** [savevm.o] Error 1

Signed-off-by: Kirill A. Shutemov <kirill@shutemov.name>
Signed-off-by: Blue Swirl <blauwirbel@gmail.com>
(cherry picked from commit 5fdb3aa16ab7628c4c3ffcbfb6861b2ad7643a6b)

Signed-off-by: Juan Quintela <quintela@redhat.com>
---
 savevm.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 savevm.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/savevm.c b/savevm.c
index 3992dfd..b3bb909 100644
--- a/savevm.c
+++ b/savevm.c
@@ -339,8 +339,7 @@ static int file_put_buffer(void *opaque, const uint8_t *buf,
 {
     QEMUFileStdio *s = opaque;
     fseek(s->stdio_file, pos, SEEK_SET);
-    fwrite(buf, 1, size, s->stdio_file);
-    return size;
+    return fwrite(buf, 1, size, s->stdio_file);
 }
 
 static int file_get_buffer(void *opaque, uint8_t *buf, int64_t pos, int size)
-- 
1.6.3.rc4.29.g8146

